src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -828,6 +833,40 @@ public static string RetrieveValueFromConnStr(string connStr, string[] keywords)
             return res;
         }
 
+        public static bool ParseDataSource(string dataSource, out string hostname, out int port, out string instanceName)
```

Comment:
I put this in DataTestUtility because this is the same code as in the SSRP PR. If this is merged, the SSRP test can be refactored to reference this shared code instead of using its own.

Reply:
Doing it in both at the same time will likely result in code conflicts. After one of these PRs is merged, the other can be updated.

Reply:
Done